Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -542,6 +543,46 @@ private static void Clean(NugetPackageIdentifier package)
             }
         }
 
+        private static bool IsAlreadyImportedInEngine(NugetPackageIdentifier package) {
+            var alreadyImportedLibs = GetAlreadyImportedLibs();
+            var isAlreadyImported = alreadyImportedLibs.Contains(package.Id);
```

Comment:
Thinking about this more, I'm wondering if this is the right level to do this. There's nothing preventing multiple assemblies being packaged in a given Nuget package. I think the more bullet proof solution would be to install the package, then check all the assemblies that have been in stalled in the package, and remove just the targeted ones.

Reply:
I'm supposing that practically this might not be the case for any of the "system" packages that we're solving for here - in which case I'm a fan of this early out. We'll need to test this out against the many different packages that have reported the issue this is solving, and if we don't run into any problems I'm definitely down to start with this implementation. =)